src/common/transformations/src/transformations/common_optimizations/sdpa_fusion.cpp

```diff
@@ -252,7 +252,24 @@ SDPAFusionMatcher::SDPAFusionMatcher() {
     auto qk_post_mask_opt_reshaped = optional<v1::Reshape>({qk_opt_scaled_opt_mask_added, any_input()});
 
     auto softmax_pred = (shape_matches("..., H, S_q, S_kv") || shape_matches("S_q, S_kv")) && consumers_count(1);
-    auto softmax = wrap_type<v8::Softmax>({qk_post_mask_opt_reshaped}, softmax_pred, {{"axis", -1}});
+    // Softmax axis can be:
+    // Pattern 1: axis = -1 (last axis)
+    // Pattern 2: axis = rank size - 1 (also means last axis for static rank inputs)
+    auto softmax_axis_neg1 = wrap_type<v8::Softmax>({qk_post_mask_opt_reshaped}, softmax_pred, {{"axis", -1}});
```

Comment:
Good catch!

I recommend to change a logic a bit:

auto axis_predicate = [](...) { // checking if the axis has  -1 and rank; you can use ov::util::try_normalize_axis to get rank value };
auto softmax_pred = axis_predicate && (shape_matches("..., H, S_q, S_kv") || shape_matches("S_q, S_kv")) && consumers_count(1);
auto softmax_axis_neg1 = wrap_type<v8::Softmax>({qk_post_mask_opt_reshaped}, softmax_pred);